--- a/src/options.c
+++ b/src/options.c
@@ -377,7 +377,7 @@
 
 #ifdef ENABLE_CHILLIREDIR
   for (i = 0; i < MAX_REGEX_PASS_THROUGHS; i++) {
-#if defined (__FreeBSD__) || defined (__APPLE__) || defined (__OpenBSD__) || defined (__NetBSD__)
+#if defined (__linux__) || defined (__FreeBSD__) || defined (__APPLE__) || defined (__OpenBSD__) || defined (__NetBSD__)
     regfree(&_options.regex_pass_throughs[i].re_host);
     regfree(&_options.regex_pass_throughs[i].re_path);
     regfree(&_options.regex_pass_throughs[i].re_qs);
--- a/src/main-redir.c
+++ b/src/main-redir.c
@@ -510,8 +510,10 @@
 
 #if defined (__FreeBSD__) || defined (__APPLE__) || defined (__OpenBSD__) || defined (__NetBSD__)
   if (!re->re_g)
+#elif defined (__linux__)
+  if (!re)
 #else
-    if (!re->allocated)
+  if (!re->allocated)
 #endif
     {
       if ((ret = regcomp(re, regex, REG_EXTENDED | REG_NOSUB)) != 0) {
