--- a/conf/down.sh.in
+++ b/conf/down.sh.in
@@ -14,6 +14,8 @@ run_down() {
     
     # site specific stuff optional
     [ -e @SYSCONFDIR@/chilli/ipdown.sh ] && . @SYSCONFDIR@/chilli/ipdown.sh
+    # OpenWRT specific
+    [ -e @SYSCONFDIR@/chilli/ipdown.lua ] && @SYSCONFDIR@/chilli/ipdown.lua
 }
 
 FLOCK=$(which flock)
--- a/conf/up.sh.in
+++ b/conf/up.sh.in
@@ -93,6 +93,8 @@ run_up() {
 
     # site specific stuff optional
     [ -e @ETCCHILLI@/ipup.sh ] && . @ETCCHILLI@/ipup.sh
+    # OpenWRT specific
+    [ -e @ETCCHILLI@/ipup.lua ] && @ETCCHILLI@/ipup.lua
 }
 
